### DISCUSSION TITLE
Why is there not a return URL parameter when you open Module Settings

### DISCUSSION THREAD

**Discussion by vnetonline** [oqtane/oqtane.framework#4005] [Ideas]
If you are in an Action when you go to Module Settings, you can't return to that action, Shouldn't there be a return URL parameter so if the user clicks `Cancel`, then they can return to the same state the module was in?

`<NavLink class="btn btn-secondary" href="@NavigateUrl()">@SharedLocalizer["Cancel"]</NavLink>`

I am referring to this line of code in `Modules\Settings.razor`

I propose that there should be a return URL so that you can return to where you came from and preserve the parameters just like `ActionLink` does.

**Comment by sbwalker** [comment]
@vnetonline This has already been changed in the Dev branch (ie. 5.1):

```
    <NavLink class="btn btn-secondary" href="@PageState.ReturnUrl">@SharedLocalizer["Cancel"]</NavLink>
```

**Comment by thabaum** [comment]
@vnetonline I suggest giving 5.1 a test drive sometime soon as we are trying to resolve issues there and have fixed so many things along the way to keep them from being "static render" problems this is a huge push coming forward, probably one of Oqtanes largest since version 3 at least...

Not trying to pile up the backlog there, but I have been reporting EVERYTHING and the community has been ON IT! :) 🚀 

5.1 issues are very welcome right now.  Please give it a test run and see if you can find issues.   Things have been updated daily for a couple months now with fixes...  I know a few of us are very dedicated now to making this 100% and only with everyone's reporting, testing, exploring can we get to this end goal.

I really appreciate you bringing up this concern, and I appreciate @sbwalker as a maintainer who approaches each challenge with a tentative demeanor, delicately navigating the complexities of engineering new technology. 

Cheers!
